tests/testthat/test-epidatacall.R

```diff
@@ -7,32 +7,21 @@ test_that("fetch_tbl", {
     time_values = epirange("2020-06-01", "2020-08-01"),
     geo_values = "ca,fl"
   )
-  # Generated with
-  # epidata_call %>%
-  # fetch_debug(format_type = "classic") %>%
-  # readr::write_rds(testthat::test_path("data/test-classic.rds"))
-  mockery::stub(fetch_classic, "httr::content", readRDS(testthat::test_path("data/test-classic.rds")))
-  mockery::stub(fetch_tbl, "fetch_classic", fetch_classic)
-  # Generated with
-  # epidata_call %>%
-  # fetch_debug(format_type = "csv") %>%
-  # readr::write_rds(testthat::test_path("data/test-csv.rds"))
-  mockery::stub(fetch_csv, "httr::content", readRDS(testthat::test_path("data/test-csv.rds")))
 
-  # This test compares the output of a tibble using fetch_tbl and fetch_csv.
-  #
-  # 1) fetch_tbl calls fetch_classic, which requests the default (classic
-  # format) from the API, uses jsonlite::fromJSON to convert the underlying data
-  # to a data.frame, and finally applies parse_data_frame is used to do the data
-  # type coersion specified by the epidata_call metadata.
-  # 2) fetch_csv requests the csv format from the API, then uses readr::read_csv
-  # to get a data.frame, and has its own methods to enforce data types.
-  tbl_out <- epidata_call %>% fetch_tbl()
-  csv_out <- epidata_call %>% fetch_csv()
-  expect_identical(tbl_out, csv_out)
+  # This test compares the output of a tibble using fetch and fetch_tbl.
+  with_mocked_bindings(
+    {
+      tbl_out <- epidata_call %>% fetch_tbl()
+      out <- epidata_call %>% fetch()
+    },
+    # RDS file generated with
+    # epidata_call %>%
+    # fetch_debug(format_type = "classic") %>%
+    # readr::write_rds(testthat::test_path("data/test-classic.rds"))
+    content = function(...) readRDS(testthat::test_path("data/test-classic.rds")),
+    .package = "httr"
```

Comment:
note: https://testthat.r-lib.org/reference/local_mocked_bindings.html says that this approach with `.package =` is "not a great idea because it affects all code in that package, not just code in your package".  I think it's probably fine here though, since we don't expect any other calls to `httr::content` besides our own direct calls.  Just noting in case you didn't already see.  This might factor into discussion [here](https://github.com/cmu-delphi/epiprocess/issues/112).

The `jsonlite::fromJSON` mock also appears to work, although I feel less certain that we wouldn't use it indirectly via `httr` vs. in the above case.  But not sure it's worth doing anything unless we actually make a decision in `::`-related discussion.

Reply:
question: do we even need this test anymore?  IIRC the old version was testing that we obtained the same content from parsing classic as from parsing csv.  But now both of these are based on the classic transfer format, and `fetch` even directly calls `fetch_tbl`.

Reply:
hm, yea this test isn't doing much. we should probably replace it with a parse function.

good point about mocking, definitely worth keeping in mind in the future.